src/algorithms/ctmrg/sparse_environments.jl

```diff
@@ -0,0 +1,265 @@
+const SparseCTMRG = CTMRG{M,true} where {M}  # TODO: Is this really a good way to dispatch on the sparse CTMRG methods?
```

Comment:
I would probably put this as a regular field. I am starting to feel less and less happy with all our type parameters, as it makes things a lot less transparent...

Reply:
> As a more general remark, I do really like the sparse environment structs, and would probably structure the code to always construct them first, and then, depending on the algorithm, either convert to a dense environment TensorMap or not. This would probably (hopefully) organise a bit of the code a bit more, such that the responsability of dealing with these objects is centralized to these structs.

When I was writing these struct I was also thinking that it would be more useful to center the CTMRG function around them. That would definitely organize the code a bit more naturally. I'll get to it next week!